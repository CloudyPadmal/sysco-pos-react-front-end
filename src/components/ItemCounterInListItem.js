import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import {
    dispatchQuickDeleteThisItemFromQuantity,
    dispatchQuickIndecrementItemFromQuantity
} from "../actions/ordercontrolactions";

/***************************************************************************************************
 *  Custom component to increment and decrement value within a minimum and
 * maximum range. A local count will be maintained to help with validation
 * @param name Identifier for this counter
 * @param min Minimum allowable value
 * @param max Maximum allowable value
 **************************************************************************************************/
class ItemCounterInListItem extends Component {

    /***********************************************************************************************
     * 
     **********************************************************************************************/
    deleteThisItem = (e) => {
        e.preventDefault();
        let P = this.props.itemsList.find((I) => I.productID === this.props.name).price;
        this.props.dispatchQuickDeleteThisItemFromQuantity(
            this.props.name, this.props.currentOrderID, this.props.count, P, this.props.passKey);
    }

    /***********************************************************************************************
     * 
     **********************************************************************************************/
    incrementItemCount = (e) => {
        e.preventDefault();
        if (this.props.count < this.props.max) {
            let P = this.props.itemsList.find((I) => I.productID === this.props.name).price;
            this.props.dispatchQuickIndecrementItemFromQuantity(
                this.props.name, true, this.props.count, this.props.currentOrderID,
                P, this.props.passKey);
        }
    }

    /***********************************************************************************************
     * 
     **********************************************************************************************/
    decrementItemCount = (e) => {
        e.preventDefault();
        if (this.props.count > this.props.min) {
            let P = this.props.itemsList.find(
                (I) => I.productID === this.props.name).price;
            this.props.dispatchQuickIndecrementItemFromQuantity(this.props.name, false,
                this.props.count, this.props.currentOrderID, P, this.props.passKey);
        }
    }

    render() {
        return (
            <React.Fragment>
                <div className="btn-toolbar" role="toolbar" aria-label="Toolbar with button groups">
                    <div className="btn-group" role="group" aria-label="Item-Group">
                        <button type="button" className="btn btn-secondary"
                            onClick={this.incrementItemCount}>&#43;</button>
                        <div style={{ width: "3em", textAlign: "center" }}>
                            <span className="input-group-text" style={{
                                display: "inline-block", borderRadius: "0px",
                                width: "100%", height: "100%"
                            }}>{this.props.count}</span></div>
                        <button type="button" className="btn btn-secondary"
                            onClick={this.decrementItemCount}>&#8722;</button>
                    </div>
                    <div className="btn-group"
                        role="group" aria-label="Delete-Group" style={{ marginLeft: "10px" }}>
                        <button
                            type="button" className="btn btn-danger"
                            onClick={this.deleteThisItem}><i className="fas fa-trash"></i></button>
                    </div>
                </div>
            </React.Fragment>
        );
    }
}

ItemCounterInListItem.propTypes = {
    dispatchQuickIndecrementItemFromQuantity: PropTypes.func.isRequired,
    dispatchQuickDeleteThisItemFromQuantity: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
    passKey: state.uac.passKey,
    itemsList: state.ord.itemsList,
    currentOrderID: state.ord.currentOrderID
});

export default connect(mapStateToProps, {
    dispatchQuickIndecrementItemFromQuantity,
    dispatchQuickDeleteThisItemFromQuantity
})(ItemCounterInListItem);
