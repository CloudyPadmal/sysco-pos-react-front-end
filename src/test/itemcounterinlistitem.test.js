import React from 'react';
import { mountWrap } from './enzyme';

import ItemCounterInListItem from '../components/ItemCounterInListItem';

describe('Item Counter In List Item', () => {

    it('displays a list of items to edit order', () => {
        const wrapper = mountWrap(<ItemCounterInListItem
            NAME={`Item`} MIN={0} MAX={34}
            COUNT={34}
            ADD_THIS_ITEM_TO_ITEMQUANTITY={() => { }}
            INDECCREMENT_ITEM_COUNT={() => { }}
            DELETE_THIS_ITEM={() => { }} />);
        expect(wrapper).toMatchSnapshot();
    });
});