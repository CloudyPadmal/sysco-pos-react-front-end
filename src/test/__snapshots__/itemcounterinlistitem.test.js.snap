// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Counter In List Item displays a list of items to edit order 1`] = `
<ItemCounterInListItem
  ADD_THIS_ITEM_TO_ITEMQUANTITY={[Function]}
  COUNT={34}
  DELETE_THIS_ITEM={[Function]}
  INDECCREMENT_ITEM_COUNT={[Function]}
  MAX={34}
  MIN={0}
  NAME="Item"
>
  <div
    aria-label="Toolbar with button groups"
    className="btn-toolbar"
    role="toolbar"
  >
    <div
      aria-label="Item-Group"
      className="btn-group"
      role="group"
    >
      <button
        className="btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <div
        style={
          Object {
            "textAlign": "center",
            "width": "3em",
          }
        }
      >
        <span
          className="input-group-text"
          style={
            Object {
              "borderRadius": "0px",
              "display": "inline-block",
              "height": "100%",
              "width": "100%",
            }
          }
        >
          34
        </span>
      </div>
      <button
        className="btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
        âˆ’
      </button>
    </div>
    <RenderDeleteButton
      COUNT={34}
      DELETE_THIS_ITEM={[Function]}
    >
      <div
        aria-label="Delete-Group"
        className="btn-group"
        role="group"
        style={
          Object {
            "marginLeft": "10px",
          }
        }
      >
        <button
          className="btn btn-danger"
          onClick={[Function]}
          type="button"
        >
          <i
            className="fas fa-trash"
          />
        </button>
      </div>
    </RenderDeleteButton>
  </div>
</ItemCounterInListItem>
`;
